[DRAFT] Migrate Elyra extensions to support JupyterLab 4.x

xinchaochen8 opened the pull request:
Fixed theme and working on ui-components

### What changes were proposed in this pull request?
Resolved type errors in Dropzone.tsx, ExpandableComponent.tsx, and ExpandableErrorDialog.tsx related to missing children property. Fixed import issues in FormEditor.tsx by addressing missing members and correcting import statements. Improved TypeScript type inference, resolved implicit 'any' types in parameters and binding elements in JSONComponent.tsx. Also, addressed TypeScript incompatibility error, clarified component structure, and ensured consistency and definition in prop types.

### How was this pull request tested?
Did not add any test.
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

xinchaochen8 committed:
Fixed issues with yarn module and build errors.

Updated LauncherModel to accommodate changes in @lumino/algorithm. Resolved build errors related to missing properties and import issues.

xinchaochen8 committed:
Type and import errors fixed in components.

Resolved type errors related to missing children property in Dropzone.tsx, ExpandableComponent.tsx, and ExpandableErrorDialog.tsx. Fixed import issues in FormEditor.tsx by addressing missing members and correcting import statements. Improved TypeScript type inference and resolved implicit 'any' types in parameters and binding elements in JSONComponent.tsx.

xinchaochen8 committed:
Fixing compile errors for ui-components

TypeScript Error: Incompatibility
Component Structure: Clarification
Prop Types: Consistency and Definition

xinchaochen8 committed:
Resolving type errors

Unable to find the documentation regarding changes that made the types unavailable, and defined new types base on available specifications in 3.6x.

`Property 'IConfig' does not exist on type 'typeof CodeEditor'.ts(2339)`
`Namespace '.../node_modules/@jupyterlab/codeeditor/lib/editor".CodeEditor' has no exported member 'IConfig'.ts(2694)`

lresende commented:
@EURICIA ok to merge this and you continue in new prs? or do you want to continue on this one ?

xinchaochen8 committed:
python editor is not able to build

Note:
to compile python editor, build script editor first

xinchaochen8 committed:
Update CodeBlock.tsx

Updated CodeBlock to uses updated version of CodeEditor.
Changed the implementation base on the source code for IOptions:
https://github.com/jupyterlab/jupyterlab/blob/d274f84/packages/codeeditor/src/editor.ts#L553

xinchaochen8 committed:
Update Tags.tsx

fixed type errors for TagsField

kchen0g6 committed:
FormRendererRegistry Changes

Changed IFormComponentRegistry in metadata and meta-data common to use IFormRendererRegistry for 4.x.

xinchaochen8 committed:
resolved build error with ui-components

The fields prop was removed from FieldTemplateProps in a recent update. Resolved the integration by passing a templates object to the fields prop. Also added validator, which is now a required prop for 5.x Form. But still need to work on resolving precise type configurations.

src: https://rjsf-team.github.io/react-jsonschema-form/docs/migration-guides/v5.x%20upgrade%20guide#rjsfcore-breaking-changes

kchen0g6 committed:
Scala editor updated

Updated Scala editor to be compatible with 4.x. The code used to use IConfig which is no longer in use. But config options is now currently under IOptions. But we need to see if It displays properly on frontend. But the code currently builds.

kchen0g6 committed:
Merge branch 'pr3201/fix' of https://github.com/EURICIA/elyra into pr3201/fix

xinchaochen8 committed:
Updated yarn.lock for @rjsf/validator-ajv8

xinchaochen8 committed:
Revert "Updated yarn.lock for @rjsf/validator-ajv8"

This reverts commit 0f9df327922a7bda20de79ef7797cbb0375bb1d1.

kchen0g6 committed:
r-editor update

Updated r-editor to be compatible with Jupyter Lab 4.x. Similar changes to scala-editor.

xinchaochen8 committed:
The fixing method componentRegistry.addRenderer

Because involved react component and does not directly conform to the IFormRenderer interface use assertion to satisfy the type requirements of the addRenderer method. Also improve types definition for imported component in ui-components. Now able to build metadata.

kchen0g6 committed:
Pipeline editor nearly fixed

I have resolved most of the errors currently in the pipeline editor however I got stuck attempting to fix the error:  error TS2322: Type '{ children: Element[]; theme: any; }' is not assignable to type 'IntrinsicAttributes & { theme: DeepPartial<Theme>; }'.
  Property 'children' does not exist on type 'IntrinsicAttributes & { theme: DeepPartial<Theme>; }'.

xinchaochen8 committed:
Refactored code snippet insertion for JupyterLab 4.x compatibility

Updated CodeSnippetDisplay class to extend MetadataDisplay with the correct props type
Replace usage of deprecated ContentFactory, but still exist issue when missing documentation.

xinchaochen8 committed:
Attempt to fix cell creation

tried to fix the first code cell creation error, but still have issue when inserting cell to notebook widget

xinchaochen8 committed:
Update CodeSnippetWidget.tsx

Just figured out how to properly add codecell.

xinchaochen8 committed:
Update CodeSnippetWidget.tsx

Fixed some method calling base on documentation.

kchen0g6 committed:
PipelineEditorFixed

Resolved final error in pipeline editor. Now it's able to build.

xinchaochen8 committed:
updated code snippet display functionality

- .value.text does not work for updating the content of Cell, Replaced with .setSource.
- Refresh no longer exists, replaced with redo. Can't find any documentation about this change.
- fixed creation of preview editor

xinchaochen8 committed:
Merge branch 'pr3201/fix' of https://github.com/EURICIA/elyra into pr3201/fix

xinchaochen8 committed:
updated naming for interface to match naming conventionn

kchen0g6 committed:
regex format match

kchen0g6 committed:
Test remove unused

kchen0g6 committed:
test

xinchaochen8 committed:
verifying elyra do build

kchen0g6 committed:
Revert "test"

This reverts commit ac10f57887c245441370069c5b04874c81aa2b8f.

kchen0g6 committed:
test

xinchaochen8 committed:
reformat some files that cause build to fail

lresende committed:
Use JupyterLab 4.0

lresende committed:
Minor dependency updates

lresende committed:
Pin jupyterlab-lsp to 5.0.3 due to lab 4.1 dependencies

xinchaochen8 committed:
test to resolve yarn failure on github

xinchaochen8 committed:
Update build.yml for GitHub workflow

xinchaochen8 committed:
Revert "Update build.yml for GitHub workflow"

This reverts commit b2630c73f4132b55c7481b7eb9d165c588c6466f.

xinchaochen8 committed:
Update build.yml for yarn workflow error

xinchaochen8 committed:
Revert "Update build.yml for yarn workflow error"

This reverts commit 9ab11ebf24aaa3114591b4d2a27c0f539e6a9797.

xinchaochen8 committed:
Update .yarnrc.yml

xinchaochen8 committed:
Update build.yml

trying to adjust workflow build issue

xinchaochen8 committed:
try latest version

xinchaochen8 committed:
maybe issue with node version?

lresende commented:
Merged, let's continue this on the Elyra pr.
